coalib/results/ResultFilter.py

```diff
@@ -0,0 +1,220 @@
+    original_origin = isinstance(original_result.origin, str) and \
+        original_result.origin or original_result.origin.__name__()
+    modified_origin = isinstance(modified_result.origin, str) and \
+        modified_result.origin or modified_result.origin.__name__()
+
+    # we cannot tolerate differences!
+    if original_origin != modified_origin:
+        return False
+
+    elif original_result.message != modified_result.message:
+        return False
+
+    elif original_result.severity != modified_result.severity:
+        return False
+
+    elif original_result.debug_msg != modified_result.debug_msg:
+        return False
+
+    else:
+        return True
+
+
+def source_ranges_match(original_file_dict,
+                        modified_file_dict,
+                        diff_dict,
+                        original_result,
+                        original_result_diff_dict,
+                        modified_result,
+                        modified_result_diff_dict):
+    """
+    Checks whether the SourceRanges of two results match
+
+    :param original_file_dict: Dict of lists of file contents before changes
+    :param modified_file_dict: Dict of lists of file contents after changes
+    :param diff_dict:          Dict of diffs describing the changes per file
+    :param original_result:    A result of the old files
+    :param original_result_diff_dict: diff for each file for this result
+    :param modified_result:    A result of the new files
+    :param modified_result_diff_dict: guess
+    :return:                     Boolean value whether the SourceRanges match
+    """
+    for file_name in original_file_dict:
+
+        try:  # fails if the affected range of the result get's modified
+            original_total_diff = \
+                diff_dict[file_name] + original_result_diff_dict[file_name]
+        except ConflictError:
+            return False
+
+        # original file with file_diff and original_diff applied
+        original_total_file = original_total_diff.modified
+        # modified file with modified_diff applied
+        modified_total_file = modified_result_diff_dict[file_name].modified
+        if original_total_file != modified_total_file:
+            return False
+    return True
+
+
+def diffs_match(original_file_dict,
+                modified_file_dict,
+                diff_dict,
+                original_result,
+                modified_result):
+    """
+    Checks whether the Diffs of two results describe the same changes
+
+    :param original_file_dict: Dict of lists of file contents before changes
+    :param modified_file_dict: Dict of lists of file contents after changes
+    :param diff_dict:          Dict of diffs describing the change in each file
+    :param original_result:    A result of the old files
+    :param modified_result:    A result of the new files
+    :return:                   Boolean value whether the Diffs match
+    """
+    return False # pragma: no cover
```

Comment:
ah interesting, I would have guessed that you need to do the pragma for the whole function... well then take it out until you need it ok?